common/transport/DcgmIpc.cpp

```diff
@@ -286,8 +288,40 @@ dcgmReturn_t DcgmIpc::InitTCPListenerSocket()
         return DCGM_ST_OK;
     }
 
+    memset(&listenAddr4, 0, sizeof(listenAddr4));
+    listenAddr4.sin_family = AF_INET;
+    listenAddr4.sin_port   = htons(m_tcpParameters.value().port);
+
+    memset(&listenAddr6, 0, sizeof(listenAddr6));
+    listenAddr6.sin6_family = AF_INET6;
+    listenAddr6.sin6_port   = htons(m_tcpParameters.value().port);
+
+    if (m_tcpParameters.value().bindIPAddress.size() > 0)
+    {
+        if (inet_aton(m_tcpParameters.value().bindIPAddress.c_str(), &listenAddr4.sin_addr) == 1)
+        {
+            memcpy(&listenAddr, &listenAddr4, sizeof(listenAddr4));
+        }
+        else if (inet_pton(AF_INET6, m_tcpParameters.value().bindIPAddress.c_str(), &listenAddr6.sin6_addr) == 1)
+        {
+            memcpy(&listenAddr, &listenAddr6, sizeof(listenAddr6));
+        }
+        else
+        {
+            DCGM_LOG_ERROR << "Unable to convert \"" << m_tcpParameters.value().bindIPAddress
+                           << "\" to a network address.";
+            return DCGM_ST_GENERIC_ERROR;
+        }
+    }
+    else
+    {
```

Comment:
In the previous code there's no handling for else part of `if(m_tcpParameters.value().bindIPAddress.size() > 0)` check. Does that mean with this commit DCGM will start supporting binding to all interfaces?

Reply:
Despite there being no else part here, the previous code made sure that we bind to all IPV4 interfaces (0.0.0.0). This was done by this line `listenAddr.sin_addr.s_addr = INADDR_ANY`

With my changes, we now bind to both IPV4 (0.0.0.0) and IPV6 (::) interfaces. I've also explicitly set IPV6_ONLY socket option to 0 to ensure this works.

Reply:
Sounds good.